### PR TITLE
update to k8s 1.30 libs and controller-runtime 0.18.4

### LINKED ISSUE #1097: Using fork of openshift/library-go
**Is this a BUG REPORT or FEATURE REQUEST?**:

> Uncomment only one, leave it on its own line:
>
/kind bug
> /kind enhancement

**What happened**:

#1091 references a fork of openshift/library-go

Once the following PR is merged, CDI should go back the mainline:

https://github.com/openshift/library-go/pull/540

**What you expected to happen**:

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
- CDI version (use `kubectl get deployments cdi-deployment -o yaml`):
- Kubernetes version (use `kubectl version`):
- Cloud provider or hardware configuration:
- Install tools:
- Others:

### PR DESCRIPTION
**Pull request by mhenriks**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #1097 

**Special notes for your reviewer**:

**Release note**:

```release-note
k8s 1.30 libs and controller-runtime 0.18.4
```

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -157,51 +159,55 @@ require (
...
+	k8s.io/sample-cli-plugin => k8s.io/sample-cli-plugin v0.30.2
+	k8s.io/sample-controller => k8s.io/sample-controller v0.30.2
 
 	kubevirt.io/containerized-data-importer-api => ./staging/src/kubevirt.io/containerized-data-importer-api
 	kubevirt.io/controller-lifecycle-operator-sdk/api => kubevirt.io/controller-lifecycle-operator-sdk/api v0.0.0-20220329064328-f3cc58c6ed90
-	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.16.3
+	sigs.k8s.io/controller-runtime => sigs.k8s.io/controller-runtime v0.18.4
 )
+
+replace kubevirt.io/controller-lifecycle-operator-sdk => github.com/mhenriks/controller-lifecycle-operator-sdk v0.0.7-0.20240702040137-0d244a67055d
```
Comment (mhenriks) [inline go.mod:213]: /hold

Until this is not pointing to my github

**Comment by akalenyu** [comment]
@awels I think something's up with FOSSA here

**Comment by akalenyu** [comment]
/approve
/cc @awels 
FOSSA is lying I think

**Comment by awels** [comment]
Hmmm so it is complaining about a depedency in vendor/github.com/hashicorp/golang-lru being MPL 2.0. Which is not compatible with our licenses. It is being vendor-ed in through github.com/openshift/library-go 
I think we should avoid anything hashicorp as much as possible.

**Comment by akalenyu** [comment]
> Hmmm so it is complaining about a depedency in vendor/github.com/hashicorp/golang-lru being MPL 2.0. Which is not compatible with our licenses. It is being vendor-ed in through github.com/openshift/library-go I think we should avoid anything hashicorp as much as possible.

right.. I see it's coming from their dep. on `distribution/distribution` in helpers we are not using:
https://github.com/openshift/library-go/pull/1529/commits/01526ee81bea21c8cb6741d34c2dbbe10ae8aa8d
anyway, I wonder how other library-go consumers are not finding this problem

**Comment by mhenriks** [comment]
@awels golang-lru code is not in vendor directory so we are not actually directly or indirectly using it. Are there exceptions for this?

**Comment by awels** [comment]
I can mark it as an exception indicating it is not being used at all. I wonder why it was flagged. You are right that it is not being vendored at all.

**Comment by akalenyu** [comment]
I think this failure is about doing a rebind when pod is still running?
`"cdi.kubevirt.io/storage.pod.phase": "Running",`
https://prow.ci.kubevirt.io/view/gs/kubevirt-prow/pr-logs/pull/kubevirt_containerized-data-importer/3336/pull-containerized-data-importer-e2e-upg/1811766743761162240

**Comment by akalenyu** [comment]
/test pull-containerized-data-importer-e2e-nfs
`Error: failed get datavolume: test-dv, client rate limiter Wait returned an error: context deadline exceeded`

**Pull request merged**